src/disk.hpp

```diff
@@ -170,7 +178,7 @@ struct FileDisk {
 
     void Write(uint64_t begin, const uint8_t *memcache, uint64_t length)
     {
-        Open();
+        Open(writeFlag | retryOpenFlag);
```

Comment:
hm. this used to rely on the default argument, which was `true`, meaning it would open the file in read-only mode.
This looks like a bug. Perhaps it was never encountered because the file would always already be open at this point.
Anyway, as far as I can tell, your patch fixes it.

Reply:
Oh, interesting. I didn't even realize that was a bug, but yeah, seems "fixed" now.